### PR TITLE
added valid_move as well as TDD for king

### PR DESCRIPTION
**Pull request by RaykPlatzek**
1. added the valid_move? method in king.rb
2. added tests for each direction to move

### CODE REVIEW

**Comment by RaykPlatzek** [comment]
@mscoutermarsh we had to make the direction checking methods public again and remove "private" because otherwise we would not have access to them. (is there another way to do it and keep them private ? )

**Comment by mscoutermarsh** [comment]
@RaykPlatzek Methods that are only used within the class should be private. So anything that's only used within `Piece`, should be private.

Methods that are called from outside of the class, such as `obstructed?` should be public.

**Comment by mscoutermarsh** [comment]
@RaykPlatzek Also! Forgot to add.

Since `King` is inheriting from `Piece`. It inherits all of the private methods as well. So it is able to use the private methods. 

(kind of confusing, let me know if doesn't make sense) 😄

File: spec/models/king_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe King, type: :model do
+  let(:game) { FactoryGirl.create(:game) }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:king) { FactoryGirl.create(:king, :is_on_board) }
```
Comment (mscoutermarsh) [inline spec/models/king_spec.rb]: really nice!

**Pull request merged**